pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -354,31 +328,58 @@ func (l *Lifecycle) Updated(obj *v3.KontainerDriver) (runtime.Object, error) {
 		return obj, l.updateDynamicSchema(obj)
 	}
 
+	var err error
+	var tmpObj runtime.Object
+	newObj := obj.DeepCopy()
+	switch {
+	case !obj.Spec.Active: // dealing with deactivate action
+		tmpObj, err = v3.KontainerDriverConditionInactive.DoUntilTrue(newObj, func() (runtime.Object, error) {
+			if err = l.dynamicSchemas.Delete(getDynamicTypeName(newObj), &v13.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+				return nil, fmt.Errorf("error deleting schema: %v", err)
+			}
+			if err = l.removeFieldFromCluster(newObj); err != nil {
+				return nil, err
+			}
+
+			RemoveCondition(newObj, v3.KontainerDriverConditionActive)
+			return newObj, nil
+		})
+		newObj = tmpObj.(*v3.KontainerDriver)
+
 	// redownload file if active AND url changed or not downloaded
 	// prevents downloading on air-gapped installations
-	var err error
-	if obj.Spec.Active && (obj.Spec.URL != obj.Status.ActualURL || v3.KontainerDriverConditionDownloaded.IsFalse(obj) || !l.driverExists(obj)) {
-		obj, err = l.download(obj)
-		if err != nil {
-			return nil, err
+	case obj.Spec.URL != obj.Status.ActualURL ||
+		v3.KontainerDriverConditionDownloaded.IsFalse(obj) ||
+		!l.driverExists(obj):
+		if !v3.KontainerDriverConditionDownloaded.IsUnknown(newObj) {
+			v3.KontainerDriverConditionDownloaded.Unknown(newObj)
+			break
 		}
+		fallthrough
+	case !v3.KontainerDriverConditionDownloaded.IsTrue(newObj):
+		newObj, err = l.download(newObj)
```

Comment:
The operation in `download` will not be executed, you can see the` condition.Once ` func:
```
func (c Cond) Once(obj runtime.Object, f func() (runtime.Object, error)) (runtime.Object, error) {
	if c.IsFalse(obj) {  ----**if it is false, return error**
		return obj, &controller.ForgetError{
			Err: errors.New(c.GetReason(obj)),
		}
	}

	return c.DoUntilTrue(obj, f)
}
```

Reply:
I didn't use the Once()